sdk/containerregistry/azure-containers-containerregistry/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,126 @@
+# Troubleshoot Azure Container Registry client library issues
+
+This troubleshooting guide contains instructions to diagnose frequently encountered issues while using the Azure Container Registry client library for Java.
+
+## General Troubleshooting
+
+Container registry service methods throw a [HttpResponseException][HttpResponseException] on failure.
+
+```java readme-sample-getProperties
+DefaultAzureCredential credential = new DefaultAzureCredentialBuilder().build();
+ContainerRepository containerRepository = new ContainerRegistryClientBuilder()
+    .endpoint(endpoint)
+    .credential(credential)
+    .buildClient()
+    .getRepository(repositoryName);
+try {
+    containerRepository.getProperties();
+} catch (HttpResponseException exception) {
+    // Do something with the exception.
+}
+```
+
+Azure Container Registry client library log errors by default. The errors and warnings in the logs or traces generally provide
+useful insights into what went wrong and sometimes include corrective actions to fix issues.
+
+Make sure logging or tracing are enabled to monitor the behavior of the application and to troubleshoot issues with Azure Container Registry library. 
+
+### Enable client logging
```

Comment:
One other thing special about ACR is the credential support - which means anybody using their own HttpPipeline (which I believe is pretty common in Java) will need to do something very similar to what we do i.e. wrap the TokenCredentials into ACRCredentials - since we are adding troubleshooting guide we may want to consider adding information on that too sometime in the future. (I had decided not to add this yet and wait for a customer ask and there wasn't any.)

Reply:
I think this would probably be a sample and maybe a section in readme, and if/once there are common issues, section in the troubleshooting guide. I'll keep it in mind and will see if there are any asks.